torch/csrc/autograd/variable.h

```diff
@@ -114,6 +123,15 @@ struct VariableImpl : public at::TensorImpl {
   friend struct VariableType;
 };
 
+struct VariableViewImpl : public VariableImpl {
+  VariableViewImpl(Variable base, at::Tensor data);
+
+  virtual std::shared_ptr<Function>& get_grad_fn() override;
+
+  Variable base;
+  int expected_version;
+};
```

Comment:
Can you add a comment briefly describing the view mechanism and mentioning which `VariableImpl` fields might be stale under what conditions?